department-of-veterans-affairs/caseflow
# Queue | Adapt AMA attorney checkout flow for RAMP appeals

Resolves #6778 

### Description
This PR updates the attorney checkout flow to support AMA tasks/appeals. It also consolidates `action`/`type` attributes between `LegacyTask`s and `Task`s, and fully deprecates the use of `issue.vacols_sequence_id` on the front-end.

### Acceptance Criteria
- [x] Get basic AMA attorney flow working for appeals where all issues are granted or denied
- [x] Verify requirements for `on_hold` tasks and ensure they're correctly filtered out for an attorney if necessary (#6750)
- [ ] ~Allow remand reasons to be chosen for AMA appeals during attorney checkout (https://github.com/department-of-veterans-affairs/caseflow/issues/5760)~
- [x] Investigate and address errors in the console that appear when navigating to `/queue` if necessary )("Same key" warning, "Invalid prop type passed to AttorneyTaskListView")
  - [x] TaskTable same key warning (#6668)
  - [x] `Invalid prop 'tasks' of type 'array' supplied to 'AttorneyTaskListView', expected 'object'`
- [ ] ~Ensure that the Notes field filled out by attorneys is saved somewhere and displayed to judges for AMA appeals~
- [x] Disable "Edit Issue", "Delete Issue", and "Add Issue" for AMA attorney checkout
- [x] If disposition is "Vacated", remove the "duplicate issue" checkbox

### Testing Plan
1. Go to `/queue` as `BVASCASPER1`, open Bob Smith's case
2. Decision Ready for Review flow (don't choose Remanded disposition)
  - If issue is Vacated, confirm "duplicate issue" checkbox remains hidden
  - Confirm Add/Edit issue links aren't displayed
3. Finish assigning case to judge (I've been using `BVANKUVALIS` because he by default has no tasks)
4. Switch to judge user, confirm Bob Smith's case appears in review task list
5. Enter Ready for Dispatch checkout flow, confirm issue dispositions are already set
